sklearn/metrics/tests/test_common.py

```diff
@@ -1201,3 +1203,53 @@ def test_no_averaging_labels():
             score_labels = metric(y_true, y_pred, labels=labels, average=None)
             score = metric(y_true, y_pred, average=None)
             assert_array_equal(score_labels, score[inverse_labels])
+
+
+@pytest.mark.parametrize(
+    'name', MULTILABELS_METRICS - NOT_SYMMETRIC_METRICS)
+def test_multilabel_label_permutations_invariance(name):
+    random_state = check_random_state(0)
+    n_samples, n_classes = 20, 4
+
+    y_true = random_state.randint(0, 2, size=(n_samples, n_classes))
+    y_score = random_state.randint(0, 2, size=(n_samples, n_classes))
+
+    metric = ALL_METRICS[name]
+    score = metric(y_true, y_score)
+
+    for perm in permutations(range(n_classes), n_classes):
+        inv_perm = np.zeros(n_classes, dtype=int)
```

Comment:
Awesome! Any reason you first do the `inv_perm` instead of directly using the permutations? Would it change the test at all if you're going through all permutations?

Reply:
Good point! This PR was updated to use the permutations directly.